ykrt/src/location.rs

```diff
@@ -49,9 +48,9 @@ pub struct Location {
     //  │             │ start compiling trace
     //  │             │ in thread
     //  │             ▼
-    //  │           ┌──────────────┐
-    //  │           │  Compiling   │
-    //  │           └──────────────┘
+    //  │           ┌──────────────┐             ┌───────────┐
+    //  │           │  Compiling   │────────────▶│  Dropped  │
+    //  │           └──────────────┘             └───────────┘
```

Comment:
Dropped means that the `HotLocation` was dropped, not the `Location` itself, right?

Should there be edges from other states into `Dropped` too? E.g. `Compiled -> Dropped`?

Reply:
> Dropped means that the HotLocation was dropped, not the Location itself, right?

They're a bit interlinked, but in this case it means the `Location` was dropped and that the `HotLocation` now has the status `Dropped`.

Reply:
And the latter query?

> Should there be edges from other states into Dropped too? E.g. Compiled -> Dropped?

Reply:
> Should there be edges from other states into Dropped too? E.g. Compiled -> Dropped?

No (at least... not yet... that might change in the future).

Reply:
OK. So when you drop a location that is already `Compiled` it just ceases to be. I think I understand.